server/backend/feedback.go

```diff
@@ -0,0 +1,142 @@
+package backend
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	pb "github.com/TUM-Dev/Campus-Backend/server/api/tumdev"
+	"github.com/TUM-Dev/Campus-Backend/server/backend/cron"
+	"github.com/TUM-Dev/Campus-Backend/server/model"
+	"github.com/gabriel-vasile/mimetype"
+	"github.com/gofrs/uuid/v5"
+	"github.com/guregu/null"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gorm.io/gorm"
+)
+
+// NewFeedback accepts a stream of feedback messages from the client and stores them in the database/file system.
+func (s *CampusServer) NewFeedback(stream pb.Campus_NewFeedbackServer) error {
+	return s.db.WithContext(stream.Context()).Transaction(func(tx *gorm.DB) error {
+		// receive metadata
+		imageCount := int32(0)
+		id, err := uuid.NewGen().NewV7()
+		if err != nil {
+			log.WithError(err).Error("Error generating uuid")
+			return status.Error(codes.Internal, "Error starting feedback submission")
+		}
+		feedback := &model.Feedback{EmailId: null.StringFrom(id.String())}
+
+		// download images
+		for {
+			req, err := stream.Recv()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				log.WithError(err).Error("Error receiving feedback")
+				deleteUploaded(tx, feedback.EmailId.String)
+				return status.Error(codes.Internal, "Error receiving feedback")
+			}
+			mergeFeedback(feedback, req)
+
+			if len(req.Attachment) > 0 {
+				imageCount = handleImageUpload(tx, &req.Attachment, imageCount, feedback.EmailId.String)
```

Comment:
Couldn't we do `handleImageUpload` once after receiving everything?  Or are we guaranteed one exact file each time the loop/recv completes? (I'm not super familiar in how this works exactly, so maybe you know from your research?)

Reply:
That is an implicite guarantee.
A stream consists of messages with a size limit of 4 MB.
- fitting more than one image into a message might be tight, but one should fit
- splitting further than 1 image/msg seemed kind of pointless to me and would have introduced a lot more bookkeeping effort (messages may by reordered in some instances)

Reply:
So the client side needs to guarantee that then? Would it be good to have it in a comment of the RPC definition?